### ISSUE TITLE
Update Vetting Status menu

### ISSUE THREAD

**Issue by dandahle** [dandahle/Catalyst-AppVetting#224]
update / change Application Vetting Status menu to better reflect new process and states 
Delete: “Application Approval Process” 
Delete: “Transferred” The change in status you made for Transferred should show up in Work Item Status. Having a separate Status category will be confusing. 
Change: “Site Assessment - Pending” to “Site Assessment” 
Change: “Approved Project” to “Application Vetted” 
Also "Handle-It" will no longer be a valid Application Status and should be deleted. However, we should make it easy to see the Handle-It work items in the application if the client comes back later for additional requests. 
![Screenshot 2021-03-25 ApplStatus-Mayberry](https://user-images.githubusercontent.com/6306417/112534911-e7ccb580-8d68-11eb-843e-5232582c0ef2.jpg)

**Comment by rarafon** [comment]
The issue with deleting a status is that any previous application with the deleted status can't have its status changed because it won't show up here. In other words, there can't be any applications with these statuses anywhere in the database.

There is still a "Site Assessment - Complete" status as well, but it only shows up for the applications that are currently in the site assessment complete, approval, or approved stage and it can't be set in the vetting worksheet. After transferring or declining the site assessment, the application will then go to the "transferred" status.

And this goes back to what I said in the emails: if you delete the "transferred" status, you still need a status for the client application to go to after its site assessment is either transferred to a project or declined. What status should that be?

**Comment by dandahle** [comment]
OK, I understand that issue. Can we just change the menu selections so they do not corrupt / change the existing database? 

The way we use "Site Assessment - Pending" and "Site Assessment Complete" provides a distinction between when information for a site assessment is gathered and when Darrell is done with his calculations and estimates for the project ("Site Assessment - Complete"). When Darrell makes the site assessment "complete", that is Shawn's signal to prepare the Project Approval Form (PAF) where he includes information about the client's situation / circumstances along with the project requirements and cost to request board approval. This is the "Project Approval" status. After the board makes a decision on the PAF then the either the status "Project Approved" or "Declined" will apply. 

All the process used to be managed in the Application View, but now is managed in the Assessment View (along with the "Project Approval" and / or "Transferred" status which I'll get to next.) 

I don't know (remember) how the database is structured, but in the Application View if we just restrict the menu choices to the new processes we are implementing (without changing the database fields / structure), that will make it less confusing to use. If not, we can just adapt manually. 

In an earlier email I think we agreed that we are talking nomenclature here - "Transferred" and "Project Approved" are in effect the same status but with a different label. We can use whatever as long as we know what it means. "Project Approval" has more meaning to the staff, but we can use "Transferred" if you like. 

Will this approach work?

**Comment by rarafon** [comment]
It wouldn't corrupt the database since the statuses aren't restricted to certain values (which is another issue that might need to be changed in the future). The only restriction is that it's a string so it can be anything right now.

It just means that any documents with the deleted statuses won't be able to be changed normally by the user. It will be fine if let's say all the applicants with these statuses were changed to something else before the new commit is pushed to the server.

There are other  alternative where you can manually change the status but I can't come up with any at the moment.

One method might be to set it up so that if there's an unrecognized status, then it can automatically changed to a certain status.


In regards to transferred, the naming of it doesn't matter to me. In your discussion about project approved, the issue is that you would also need to include the declined site assessments somewhere, which is why I wanted to know what that would be. 

In fact, this could even be the default application status to go to if there's an unrecognized status.

**Comment by dandahle** [comment]
Andy, what is the status for this issue? Do we need to talk it through?

**Comment by rarafon** [comment]
Please see what I said in the last comment. It will affect old code and data, and there's a strong possibility that it will introduce bugs to the old code.

> Change: “Site Assessment - Pending” to “Site Assessment”
Site Assessment - Complete is still being used. It's just not selectable / changeable in the vetting page. It has to be done in the assessment page (by either declining or transferring the assessment after the applicant enters the assessment complete stage).

**Comment by rarafon** [comment]
Ideally, you would want to ensure that all the applicants do not have the status that you want to to remove before removing it.

There are ways to work around it, but I can't think of any at the moment probably due to fatigue.

**Comment by dandahle** [comment]
We can learn to live with this as is for now to get started with the tool in production. However, I'm going to leave it open for now in case we get a new team that can look into some of the issues Andy brought up about unrestricted strings being used.

**Comment by rarafon** [comment]
I deleted project approval, application approval, and  handle-it and added application vetted.

I kept transferred, but this can be changed to another name. The container itself is important in how the applications are directed.

I kept site assessment pending since there is a site assessment complete that will show when the application after has the  assessment completed and until it is transferred. During this phase, the document status can't be changed in vetting.

I had to make changes in multiple files, old and new, so there might still be some bugs.
